### ISSUE TITLE
Including checker framework annotations in auto-value breaks checkerframework >= 2.9.0

### ISSUE THREAD

**Issue by rkraneis** [google/auto#749]
We use checkerframework as well as auto-value (1.6.5) in our maven-based build. After upgrading to version 2.9.0 of checkerframework, we had to investigate
```
[INFO] -------------------------------------------------------------
[WARNING] COMPILATION WARNING : 
[INFO] -------------------------------------------------------------
[WARNING] Cannot find annotation method 'types()' in type 'org.checkerframework.framework.qual.DefaultFor'
[INFO] 2 warnings 
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] InvocationTargetException when invoking constructor for class org.checkerframework.checker.nullness.KeyForAnnotatedTypeFactory; Underlying cause: java.lang.NoSuchMethodError: org.checkerframework.framework.qual.DefaultFor.typeKinds()[Lorg/checkerframework/framework/qual/TypeKind;
  Exception: java.lang.reflect.InvocationTargetException; Stack trace: sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
```
This seems to be caused by auto-value shading an older version of checkerframework.

Version 2.9.0 of checkerframwork add the two mentioned methods `types()` and `typeKinds()` to the `DefaultFor` annotation.
_Edit:_ we run both annotation processors by depending on the processors as `provided` dependencies and relying on SPI discovery.
_Edit 2:_ switching both annotation processors over to `annotationProcessorPaths` (and not having them as `provided` dependencies) does **not** resolve the problem.
_Edit 3:_ changing the order of declarations from _auto-value, checker_ to _checker, auto-value_ resolves the problem in an MWE but seems quite fragile, especially when depending on `provided` dependencies.

**Comment by eamonnmcmanus** [comment]
Could you add the complete stack trace of the exception?

**Comment by rkraneis** [comment]
```
InvocationTargetException when invoking constructor for class org.checkerframework.checker.nullness.KeyForAnnotatedTypeFactory; Underlying cause: java.lang.NoSuchMethodError: org.checkerframework.framework.qual.DefaultFor.typeKinds()[Lorg/checkerframework/framework/qual/TypeKind;
  Exception: java.lang.reflect.InvocationTargetException; Stack trace: sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
  sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
  java.lang.reflect.Constructor.newInstance(Constructor.java:423)
  org.checkerframework.common.basetype.BaseTypeChecker.invokeConstructorFor(BaseTypeChecker.java:274)
  org.checkerframework.common.basetype.BaseTypeVisitor.createTypeFactory(BaseTypeVisitor.java:260)
  org.checkerframework.common.basetype.BaseTypeVisitor.<init>(BaseTypeVisitor.java:219)
  org.checkerframework.common.basetype.BaseTypeVisitor.<init>(BaseTypeVisitor.java:208)
  org.checkerframework.common.basetype.BaseTypeChecker.createSourceVisitor(BaseTypeChecker.java:223)
  org.checkerframework.common.basetype.BaseTypeChecker.createSourceVisitor(BaseTypeChecker.java:84)
  org.checkerframework.framework.source.SourceChecker.initChecker(SourceChecker.java:895)
  org.checkerframework.common.basetype.BaseTypeChecker.initChecker(BaseTypeChecker.java:99)
  org.checkerframework.common.basetype.BaseTypeChecker.initChecker(BaseTypeChecker.java:90)
  org.checkerframework.framework.source.SourceChecker.typeProcessingStart(SourceChecker.java:851)
  org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:157)
  com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:681)
  com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:111)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1342)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
  com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
  com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
  com.sun.tools.javac.main.Main.compile(Main.java:523)
  com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
  com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
  org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:126)
  org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:174)
  org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1134)
  org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:187)
  org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
  org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
  org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
  org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
  org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
  org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
  org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
  org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
  org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
  org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
  org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
  org.apache.maven.cli.MavenCli.execute(MavenCli.java:954)
  org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
  org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
  sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  java.lang.reflect.Method.invoke(Method.java:498)
  org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
  org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
  org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
  org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
  Underlying Exception: java.lang.NoSuchMethodError: org.checkerframework.framework.qual.DefaultFor.typeKinds()[Lorg/checkerframework/framework/qual/TypeKind;; Stack trace: org.checkerframework.framework.type.typeannotator.DefaultForTypeAnnotator.<init>(DefaultForTypeAnnotator.java:71)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.createTypeAnnotator(GenericAnnotatedTypeFactory.java:372)
  org.checkerframework.framework.type.GenericAnnotatedTypeFactory.postInit(GenericAnnotatedTypeFactory.java:237)
  org.checkerframework.checker.nullness.KeyForAnnotatedTypeFactory.<init>(KeyForAnnotatedTypeFactory.java:72)
  sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
  sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
  sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
  java.lang.reflect.Constructor.newInstance(Constructor.java:423)
  org.checkerframework.common.basetype.BaseTypeChecker.invokeConstructorFor(BaseTypeChecker.java:274)
  org.checkerframework.common.basetype.BaseTypeVisitor.createTypeFactory(BaseTypeVisitor.java:260)
  org.checkerframework.common.basetype.BaseTypeVisitor.<init>(BaseTypeVisitor.java:219)
  org.checkerframework.common.basetype.BaseTypeVisitor.<init>(BaseTypeVisitor.java:208)
  org.checkerframework.common.basetype.BaseTypeChecker.createSourceVisitor(BaseTypeChecker.java:223)
  org.checkerframework.common.basetype.BaseTypeChecker.createSourceVisitor(BaseTypeChecker.java:84)
  org.checkerframework.framework.source.SourceChecker.initChecker(SourceChecker.java:895)
  org.checkerframework.common.basetype.BaseTypeChecker.initChecker(BaseTypeChecker.java:99)
  org.checkerframework.common.basetype.BaseTypeChecker.initChecker(BaseTypeChecker.java:90)
  org.checkerframework.framework.source.SourceChecker.typeProcessingStart(SourceChecker.java:851)
  org.checkerframework.javacutil.AbstractTypeProcessor$AttributionTaskListener.finished(AbstractTypeProcessor.java:157)
  com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:681)
  com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:111)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1342)
  com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
  com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
  com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
  com.sun.tools.javac.main.Main.compile(Main.java:523)
  com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
  com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
  org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:126)
  org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:174)
  org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1134)
  org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:187)
  org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
  org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
  org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
  org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
  org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
  org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
  org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
  org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
  org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
  org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
  org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
  org.apache.maven.cli.MavenCli.execute(MavenCli.java:954)
  org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
  org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
  sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  java.lang.reflect.Method.invoke(Method.java:498)
  org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
  org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
  org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
  org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
```

**Comment by rkraneis** [comment]
![image](https://user-images.githubusercontent.com/1737287/63149133-92b0b580-c003-11e9-8b0b-e798f0e3726e.png)

**Comment by rkraneis** [comment]
![image](https://user-images.githubusercontent.com/1737287/63149203-bc69dc80-c003-11e9-8163-c19057010940.png)

**Comment by rkraneis** [comment]
https://github.com/rkraneis/autovaluechecker

**Comment by rkraneis** [comment]
```
$ javac -version
javac 1.8.0_212
```

**Comment by eamonnmcmanus** [comment]
Sorry, I was being obtuse. The issue is that AutoValue pulls in `org.checkerframework` via its dependencies, and the copy in the shipped jar is not shaded as it should be. We can fix that in the next release.

**Comment by eamonnmcmanus** [comment]
This is fixed in AutoValue 1.6.6.
